File: typescript/src/initiate_vehicle_signup.ts
import { createNodeClient } from "./client"
import {create} from "@bufbuild/protobuf";
import {
  InitiateVehicleSignUpRequestSchema
} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb";

const client = createNodeClient()

const request = create(InitiateVehicleSignUpRequestSchema, {
  email: "sujith@compassiot.com.au",
  vin: "LRW3F7FS9PC893577"
})

await client.initiateVehicleSignUp(request)

File: python/initiate_vehicle_signup.py
from client import create_native_client
import nativeconnect.api.v1.nativeconnect_pb2 as native


def main():
    client = create_native_client()

    request = native.InitiateVehicleSignUpRequest(
        vin="LRW3F7FS9PC893577",
        email="sujith@compassiot.com.au"
    )

    response = client.InitiateVehicleSignUp(request)

    print(response)


if __name__ == "__main__":
    main()
